package/cloudshell/cm/ansible/ansible_shell.py

```diff
@@ -112,14 +116,15 @@ def _add_host_vars_files(self, ansi_conf, logger):
 
     def _download_playbook(self, ansi_conf, cancellation_sampler, logger):
         """
+        :param AnsibleConfiguration ansi_conf
         :type ansi_conf: AnsibleConfiguration
         :type cancellation_sampler: CancellationSampler
         :type logger: Logger
         :rtype str
         """
         repo = ansi_conf.playbook_repo
         auth = None
-        if ansi_conf.playbook_repo.username or ansi_conf.playbook_repo.token:
+        if ansi_conf.playbook_repo.username or ansi_conf.playbook_repo.token or ansi_conf.playbook_repo.password:
```

Comment:
what is purpose here?
can there be a password if no username?

Reply:
there can be a token with no username. My use case here was backwards support of driver for users on system without the token attribute field available yet. If the token field is empty fallback to the password.  Customers can use the driver and won't have to migrate attributes from password to token.